.github/workflows/node.yml

```diff
@@ -128,52 +175,74 @@ jobs:
-
-    #         - name: Install NodeJS
-    #           run: |
-    #               brew update
-    #               brew upgrade || true
-    #               brew install node
-
-    #         - name: Downgrade npm major version to 8
-    #           run: |
-    #               npm i -g npm@8
-
-    #         - name: Build Node wrapper
-    #           uses: ./.github/workflows/build-node-wrapper
-    #           with:
-    #               os: "macos"
-    #               named_os: "darwin"
-    #               arch: "arm64"
-    #               target: "aarch64-apple-darwin"
-    #               github-token: ${{ secrets.GITHUB_TOKEN }}
-    #               engine-version: "7.2.5"
-
-    #         - name: Test compatibility
-    #           run: npm test -- -t "set and get flow works"
-    #           working-directory: ./node
-
-    #         - name: Upload test reports
-    #           if: always()
-    #           continue-on-error: true
-    #           uses: actions/upload-artifact@v4
-    #           with:
-    #               name: test-report-node-${{ matrix.engine.type }}-${{ matrix.engine.version }}-macos
-    #               path: |
-    #                   node/test-report*.html
-    #                   utils/clusters/**
-    #                   benchmarks/results/**
+    build-node-macos:
+        needs: get-matrices
+        strategy:
+            fail-fast: false
+            matrix:
+                host:
+                    [
+                        {
+                            "RUNNER": "macos-latest",
+                            "OS": "macos",
+                            "ARCH": "arm64",
+                            "TARGET": "aarch64-apple-darwin",
+                        },
+                        {
+                            "RUNNER": "macos-12",
+                            "OS": "macos",
+                            "ARCH": "x64",
+                            "TARGET": "x86_64-apple-darwin",
+                        },
+                    ]
+                engine: ${{ fromJson(needs.get-matrices.outputs.engine-matrix-output) }}
+        if: (github.repository_owner == 'valkey-io' && github.event_name == 'schedule') || github.event.label.name == 'Core changes'
+        runs-on: ${{ matrix.host.RUNNER }}
+        timeout-minutes: 25
+        steps:
+            - uses: actions/checkout@v4
+              with:
+                  submodules: recursive
+            - name: Set up Homebrew
+              uses: Homebrew/actions/setup-homebrew@master
+
+            - name: Install NodeJS
+              run: |
+                  brew update
+                  brew upgrade || true
+                  brew install node
+
+            - name: Downgrade npm major version to 8
+              run: |
+                  npm i -g npm@8
+
+            - name: Build Node wrapper
+              uses: ./.github/workflows/build-node-wrapper
+              with:
+                  os: "macos"
+                  named_os: "darwin"
```

Comment:
why aren't you using the values from matrix.host?

Reply:
In this case it's always macOS, why make it prone to error if using the external values is not necessary?